homeassistant/components/waqi/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""
+Support for the World Air Quality Index service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/waqi/
+"""
+import asyncio
+import logging
+from datetime import timedelta
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.const import (CONF_TOKEN)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['waqiasync==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Data provided by the World Air Quality Index project'
+
+CONF_LOCATIONS = 'locations'
+CONF_STATIONS = 'stations'
+
+DOMAIN = 'waqi'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+TIMEOUT = 10
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_STATIONS): cv.ensure_list,
+        vol.Required(CONF_TOKEN): cv.string,
+        vol.Required(CONF_LOCATIONS): cv.ensure_list
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the WAQI component."""
+    import waqiasync
+
+    conf = config.get(DOMAIN)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.